src/app/features/wallets/wallets.page.ts

```diff
@@ -74,9 +97,58 @@ export class WalletsPage {
 
   async refreshBalance(event: Event) {
     (<CustomEvent>event).detail.complete();
-    this.isLoadingBalance$.next(true);
-    await this.diaBackendWalletService.syncAssetWalletBalance$().toPromise();
-    this.isLoadingBalance$.next(false);
+    this.networkConnected$
+      .pipe(
+        first(),
+        concatMap(networkConnected => {
+          if (!networkConnected) {
+            return this.errorService.toastError$(
+              this.translocoService.translate(
+                `error.wallets.noNetworkConnectionCannotRefreshBalance`
+              )
+            );
+          }
+          this.isLoadingBalance$.next(true);
+          return forkJoin([
+            this.diaBackendWalletService.syncAssetWalletBalance$(),
+            this.diaBackendAuthService.syncProfile$(),
+          ]).pipe(
+            tap(async ([assetWallet, _]) => {
+              this.totalBalance$.next(
+                Number(assetWallet[1]) +
+                  (await this.diaBackendAuthService.getPoints())
+              );
+            }),
```

Comment:
Most changes in `refreshBalance` and `onDepositWithdrawBtnClick` here are the same as in https://github.com/numbersprotocol/capture-lite/pull/1374 except the highlted lines here.